### ISSUE TITLE
Rule proposal: `no-unnecessary-length-check-before-some-every`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#897]
I havn't think through the details, but cases:

## Fail

```js
if (array.length === 0 || array.every(Boolean)) {}
```

```js
if (array.length !== 0 && array.some(Boolean)) {}
```

## Pass

```js
if (array.every(Boolean)) {}
```

```js
if (array.some(Boolean)) {}
```

```js
if (array.length !== 0 && array.every(Boolean)) {}
```

```js
if (array.length === 0 || array.some(Boolean)) {}
```

**Comment by sindresorhus** [comment]
How about `Array#map` or such check before a loop?

**Comment by sindresorhus** [comment]
Why should `if (array.length !== 0 && array.every(Boolean)) {}` pass?

**Comment by fisker** [comment]
To ensure `array` is not empty.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fisker** [comment]
Let's name it `no-unnecessary-length-check`? the original one seem too long.

**Comment by fisker** [comment]
Or, should we name it `no-useless-length-check`?
